cardstack/src/screens/Backup/BackupSeedPhraseConfirmationScreen/useBackupSeedPhraseConfirmationScreen.ts

```
@@ -0,0 +1,75 @@
+import { useNavigation, useRoute } from '@react-navigation/native';
+import { useCallback, useRef, useState, useMemo } from 'react';
+
+import { useWalletManualBackup } from '@cardstack/hooks/backup/useWalletManualBackup';
+import { Routes } from '@cardstack/navigation';
+import { RouteType } from '@cardstack/navigation/types';
+
+import { shuffleSeedPhraseAsArray } from './utils';
+
+interface NavParams {
+  seedPhrase: string;
+}
+
+// Selection UI is hard-coded for 12 words.
+const SEED_PHRASE_LENGTH = 12;
+
+export const useBackupSeedPhraseConfirmationScreen = () => {
+  const {
+    params: { seedPhrase },
+  } = useRoute<RouteType<NavParams>>();
+
+  const { navigate } = useNavigation();
+
+  const { confirmBackup } = useWalletManualBackup();
+
+  const selectedWordsIndexes = useRef<number[]>([]).current;
+  const [selectedWords, setSelectedWords] = useState<string[]>([]);
+
+  const shuffledWords = useMemo(() => shuffleSeedPhraseAsArray(seedPhrase), [
+    seedPhrase,
+  ]);
+
+  const selectedSeedPhraseAsString = useMemo(() => selectedWords.join(' '), [
+    selectedWords,
+  ]);
+
+  const isSelectionComplete = useMemo(
+    () => selectedWords.length === SEED_PHRASE_LENGTH,
+    [selectedWords]
+  );
+
+  const isSeedPhraseCorrect = useMemo(
+    () => seedPhrase === selectedSeedPhraseAsString,
+    [seedPhrase, selectedSeedPhraseAsString]
+  );
+
+  const handleWordPressed = useCallback(
+    (index: number) => {
+      selectedWordsIndexes.push(index);
+      setSelectedWords([...selectedWords, shuffledWords[index]]);
+    },
+    [shuffledWords, selectedWords, selectedWordsIndexes, setSelectedWords]
+  );
+
+  const handleConfirmPressed = useCallback(() => {
+    confirmBackup();
+    navigate(Routes.WALLET_SCREEN);
```

nit: the confirm backup has a try/catch block, but we are navigating, either way, without knowing if it's successful or not, maybe we could have a onSuccess callback, idk, for the manual backup the error risk is small, so just throwing the idea, maybe a todo

---

This can be tricky. I think the hook handling the exception is enough, it reports to sentry so we can analyse the error.
Maybe we could show an error alert as well? Not sure.

Also, instead of a success callback, it's better to just raise the exception up.